woocommerce/woocommerce-gutenberg-products-block
# Product Collection: Add `Attributes` filter control to sidebar settings

This PR adds `Attributes` filter control to sidebar settings.
<img width="266" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/16707866/a7aac444-d01d-40c1-8e0d-66883d083e4c">

Fixes #9363 

### Testing

1. Create a new post
2. Add Product Collection block
3. With the `Product Collection` block selected, inspect the sidebar for the `Product attributes` control. Confirm its appearance and functionalities:
   - The control should be displayed as shown in the images below:
<img width="280" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/16707866/459ffec4-5276-4a95-a3ca-3e4f6762bea9">
<img width="266" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/16707866/a7aac444-d01d-40c1-8e0d-66883d083e4c">

4. Test the `Product attributes` control by selecting any attributes. For example, color -> blue:
   - After the selection, verify that changes are immediately visible in the Editor view.

5. Save and publish the post:
   - Click on `Publish...` at the top right corner.
   - After publishing, navigate to the Frontend and verify that the changes you made in the Editor are accurately reflected.

6. Return to the Editor and validate the `Reset all` functionality of the filters:
   - Make sure that clicking the `Reset all` button resets your selected filters.
   - The `Reset all` button can be found as shown in the image below:
<img width="270" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/16707866/fa14b1c9-3db9-4743-af97-a543b7e9a90f">

7. Test the `Manage attributes` functionality:
   - Click on the `Manage attributes` link.
   - Verify that this redirects you to the designated page where you can manage product attributes



### WooCommerce Visibility




### Changelog

> Product Collection: Add `Attributes` filter control to sidebar settings
